_posts/2020-07-21-ios-tracker-1-4-0-released/index.md

```diff
@@ -0,0 +1,103 @@
+
+1. [Avoid duplicate event IDs when reusing event objects](#duplicates)
+2. [Support for single track method (similar to Android)](#singletrack)
+3. [Support for setting the true timestamp](#timestamp)
+4. [Updates and bug fixes](#updates)
+5. [Documentation](#documentation)
+6. [Getting help](#help)
+
+<!--more-->
+
+<h2 id="duplicates">1. Avoid duplicate event IDs when reusing event objects</h2>
+
+In previous versions, the event objects created by the user already contained the `event_id`. If an event object is then tracked multiple times, i.e. 
+
+{% highlight objc %}
+SPScreenView *event = [SPScreenView build:^(id builder) {
+    [builder setName:@"Home screen"];
+    [builder setType:@"Navigation bar"];
+    [builder setTransitionType:@"swipe"]
+}];
+
+[tracker track:event];
+[tracker track:event];
+{% endhighlight %}
+
+this will lead to multiple events with the same event ID and device created timestamp being sent to the collector [GitHub issues #521](https://github.com/snowplow/snowplow-objc-tracker/issues/521). Furthermore, this can also lead to duplication of contexts in the same tracked event [GitHub issue #524](https://github.com/snowplow/snowplow-objc-tracker/issues/524). 
+
+Version 1.4.0 moves the assignment of event ID into the event processing, so that the event object no longer contains it. This means you can now track the same event object multiple times, and each event will have a unique event ID and contexts will not be duplicated.
+
+<h2 id="singletrack">2. Support for a single track method (similar to Android)</h2>
+
+Snowplow's Android tracker uses a single `track` method for all the events tracked. To date, the iOS tracker has multiple specific track methods. However, as each event has a specific object type, sending each specific object type to a specific track method is redundant. Therefore we have simplified the tracker and it now uses a single `track` method for all events as well [GitHub issue #518](https://github.com/snowplow/snowplow-objc-tracker/issues/518).
+
+For example, previously a screen view would need to be tracked manually like so:
+
+{% highlight objc %}
+SPScreenView *event = [SPScreenView build:^(id builder) {
+    [builder setName:@"Home screen"];
+    [builder setType:@"Navigation bar"];
+    [builder setTransitionType:@"swipe"]
+}];
+
+[tracker trackScreenViewEvent:event];
+{% endhighlight %}
+
+Now, it can simply be tracked like this: 
+
+{% highlight objc %}
+SPScreenView *event = [SPScreenView build:^(id builder) {
+    [builder setName:@"Home screen"];
+    [builder setType:@"Navigation bar"];
+    [builder setTransitionType:@"swipe"]
+}];
+
+[tracker track:event];
+{% endhighlight %}
+
+<h2 id="timestamp">3. Support for setting the true timestamp</h2>
+
+Version 1.4.0 also comes with support for the `true_tstamp` [GitHub issue #276](https://github.com/snowplow/snowplow-objc-tracker/issues/276), an optional timestamp that can be manually set like so:  
+
+{% highlight objc %}
+
```

Comment:
Hey @AlexBenny could you provide the code for how the true timestamp can be set? I couldn't find it in the docs..

Reply:
👍